app/experimenter/experiments/forms.py

```diff
@@ -570,13 +589,50 @@ class ExperimentObjectivesForm(ChangeLogMixin, forms.ModelForm):
         widget=forms.Textarea(attrs={"class": "form-control", "rows": 20}),
     )
 
+    survey = forms.TypedChoiceField(
+        label=Experiment.SURVEY_REQUIRED_LABEL,
+        help_text=Experiment.SURVEY_HELP_TEXT,
+        choices=RADIO_OPTIONS,
+        widget=RadioWidgetCloser,
+        coerce=coerce_truthy,
+        empty_value=None,
+    )
+    survey_urls = forms.CharField(
+        required=False,
+        help_text=Experiment.SURVEY_HELP_TEXT,
+        label="Survey URLs",
+        widget=forms.Textarea(attrs={"class": "form-control", "rows": 1}),
+    )
+    survey_instructions = forms.CharField(
+        required=False,
+        label=Experiment.SURVEY_INSTRUCTIONS_LABEL,
+        help_text=Experiment.SURVEY_LAUNCH_INSTRUCTIONS_HELP_TEXT,
+        widget=forms.Textarea(attrs={"class": "form-control", "rows": 10}),
+    )
+
     class Meta:
         model = Experiment
-        fields = ("objectives", "analysis_owner", "analysis")
+        fields = (
+            "objectives",
+            "analysis_owner",
+            "analysis",
+            "survey",
+            "survey_urls",
+            "survey_instructions",
+        )
 
+    def clean(self):
+        # if the survey radio button is false, we make sure to not
+        # save any information for urls or instructions. we also want
+        # to make sure that if they once answered yes, and saved the
+        # form, but now are removing the survey, that we clear out the
+        # database of the url and instructions field
```

Comment:
I understand your logic here, but this isn't generally a pattern I think we adopt in other places, ie clearing out fields.  I *think* that could lead to confusing interactions/behaviours.  I would suggest rather than clearing out the fields, we only show the survey information if the survey yes/no field is set to yes.

So then it would look like, surveys required: yes. urls/info: etcetc.  Survey section is shown.  Go back and change survey required: no. info is hidden on detail page.  Oops it is still required, go back and change back to yes, info shows back up again.

So then on the detail page you just check that the survey required field is true before showing the info.

Does that seem sensible @uhlissuh ?

Reply:
Yes totally. I took out the logic that was clearing out the database after a switch from Yes to No.